gson/src/main/java/com/google/gson/JsonDeserializer.java

```diff
@@ -51,6 +51,8 @@
  * class IdDeserializer implements JsonDeserializer&lt;Id&gt;() {
  *   public Id deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
  *       throws JsonParseException {
+ *         JsonObject jsonObject = jsonElement.getAsJsonObject();
```

Comment:
Could you please adjust the indentation so that it is the same as for the `return` statement below?

It looks like this example is still not functional; you added a variable `long idValue` but the line below still references an unknown variable `id` and calls a method `getValue()` on it.

Reply:
updated